src/sentry/utils/sdk.py

```diff
@@ -104,16 +104,16 @@ def is_current_event_safe():
     Tests the current stack for unsafe locations that would likely cause
     recursion if an attempt to send to Sentry was made.
     """
+    scope = Scope.get_isolation_scope()
 
-    with configure_scope() as scope:
-        # Scope was explicitly marked as unsafe
-        if scope._tags.get(UNSAFE_TAG):
-            return False
+    # Scope was explicitly marked as unsafe
+    if scope._tags.get(UNSAFE_TAG):
```

Comment:
I am unsure whether this code works as expected now that we have current scopes as well

Reply:
Although, it is out of scope for this PR to fix this here, I am wondering whether we should maybe make an issue for this?

Reply:
I think as long as the tag is explicitly set on the isolation scope and then also retrieved from the isolation scope, this should work ok after your changes